src/lock/mod.rs

```diff
@@ -74,6 +74,13 @@ pub fn config(ctx: &Context, config: Config) -> Result<LockedConfig> {
                 Plugin::External(plugin) => Either::Left((index, plugin)),
                 Plugin::Inline(plugin) => Either::Right((index, LockedPlugin::Inline(plugin))),
             });
+    let inlines: Vec<_> = inlines
+        .into_iter()
+        .filter(|(_, p)| match p {
+            LockedPlugin::External(_) => true,
+            LockedPlugin::Inline(p) => p.matches_profile(ctx),
+        })
+        .collect();
```

Comment:
You can prevent the collect here since we are just chaining this below. Also suggest only convert the inline plugin to LockedPlugin after filtering.

```suggestion
    let inlines = inlines.into_iter().filter_map(|p| p.matches_profile(ctx).then(LockedPlugin::Inline));
```